src/core/lib/security/credentials/ssl/ssl_credentials.c

```diff
@@ -157,6 +196,65 @@ static void ssl_build_server_config(
     config->pem_key_cert_pairs[i].private_key =
         gpr_strdup(pem_key_cert_pairs[i].private_key);
   }
+
+  return config;
+}
+
+void grpc_ssl_server_certificate_config_release(
+    grpc_ssl_server_certificate_config *config) {
+  size_t i;
+
+  if (config == NULL) return;
+
+  for (i = 0; i < config->num_key_cert_pairs; i++) {
+    gpr_free((void *)config->pem_key_cert_pairs[i].private_key);
+    gpr_free((void *)config->pem_key_cert_pairs[i].cert_chain);
+  }
+  gpr_free(config->pem_key_cert_pairs);
+  gpr_free(config->pem_root_certs);
+}
+
+grpc_ssl_server_credentials_options *
+grpc_ssl_server_credentials_create_options_using_config(
+    grpc_ssl_client_certificate_request_type client_certificate_request,
+    grpc_ssl_server_certificate_config *config) {
+  grpc_ssl_server_credentials_options *options;
+
+  if (config == NULL) {
+    gpr_log(GPR_ERROR, "Certificate config must not be NULL.");
+    return NULL;
+  }
+
+  options = gpr_zalloc(sizeof(grpc_ssl_server_credentials_options));
+  options->client_certificate_request = client_certificate_request;
+  options->certificate_config = grpc_ssl_server_certificate_config_create(
```

Comment:
nit: use a `config_2 = config_dup(config_1)` function?

Reply:
Other than name, would there be an implementation difference with your suggestion? If so, I'm missing it.

Reply:
nope, not materially different; just that it's logically duplicating the config. so for, example, if the config struct ever changes, i.e., requires more or fewer args to create, then you'd just change the dup function, and not have to update the call here....
but yea it's just a nit, not super important :)

Reply:
If that were a public API I would definitely agree with @cauthu about a `_dup` method. However, this is just internal code.